pkg/network/controllers/vm_test.go

```diff
@@ -244,6 +244,57 @@ var _ = Describe("VM Network Controller", func() {
 		Expect(iface.State).NotTo(Equal(v1.InterfaceStateAbsent))
 	})
 
+	DescribeTable("sync updates link state correctly", func(fromState, toState v1.InterfaceState, expectChange bool) {
+		clientset := fake.NewSimpleClientset()
+		c := controllers.NewVMController(
+			clientset,
+			stubPodGetter{pod: &k8sv1.Pod{}},
+		)
+
+		vmi := libvmi.New(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+
+		vmi.Status.Interfaces = append(vmi.Status.Interfaces, v1.VirtualMachineInstanceNetworkInterface{Name: "default"})
+
+		vm := libvmi.NewVirtualMachine(vmi.DeepCopy())
+
+		// Simulate the existence of the VMI on the server, with a different iface state (to allow the Sync to patch it).
+		*vmiLinkStateIface0(vmi) = fromState
+		_, err := clientset.KubevirtV1().VirtualMachineInstances(vmi.Namespace).Create(context.Background(), vmi, k8smetav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		*vmLinkStateIface0(vm) = toState
+		if expectChange {
```

Comment:
Please pass the expected link-state and not a bool.
We prefer no conditions in the tests and in this case it is simple.

We actually want to see to what state it changed or not.